MystenLabs/sui
# [ts-sdk] Implement getOwnedObjects

This PR sets up the JsonRPC Client and queries `getOwnedObjects` endpoint in the RPC server. The SDK also validates the data from the server using https://docs.superstructjs.org/


## Unit Testing

- Added MockServer
- Added `sdk/typescript/test/rpc/client.test.ts`

## Manual Testing
1. create a react component like the following

```
import React, { useCallback } from "react";
import { JsonRpcProvider } from "sui.js";

const endpoint = "http://127.0.0.1:5001/";
const address = "C5206DD02C86A510C4848516229B02ADDFACBE55";

export function ObjectsDemo() {
  const handleSubmit = useCallback(
    async (e: React.FormEvent<HTMLFormElement>) => {
      e.preventDefault();
      const provider = new JsonRpcProvider(endpoint);
      const resp = await provider.getOwnedObjectRefs(address);
      console.log(resp);
    },
    []
  );

  return (
    <form onSubmit={handleSubmit}>
      <label>Get Objects</label>
      <input type="submit" value="Get" />
    </form>
  );
}
```
2. Use [npm link](https://medium.com/dailyjs/how-to-use-npm-link-7375b6219557) to link sui.js to the React App
3. run `./rpc_server`
4. Observe the following in console when clicking the Get button

```
[
    {
        "objectId": "8dc6a6f70564e29a01c7293a9c03818fda2d049f",
        "version": 0,
        "digest": "CI8Sf+t3Xrt5h9ENlmyR8bbMVfg6df3vSDc08Gbk9/g="
    },
    {
        "objectId": "abd4f956ef290ceb80cc257666e1de5c1f7a9127",
        "version": 0,
        "digest": "GMJfeKt0BSyIM5wLXlg3yN7MTMarmgnWSnTqTlCpTr4="
    },
    {
        "objectId": "ae7a39eb7c89bc59fa7587cd30d533ca67fb6182",
        "version": 0,
        "digest": "5CpX1G81xW6hoMPtjlcyiJQKw1/dc869KTo0PnPzFV0="
    },
    {
        "objectId": "c7ad4a1d9f773774d5fc3fbfa07011b9e883f67d",
        "version": 0,
        "digest": "FYkPySLjQPkKNFn+XWGalRZDF2oKKqfkTEzkBH0By18="
    },
    {
        "objectId": "e8dfc7e7f71262b2c6a5bddb8268ae38a507c53e",
        "version": 0,
        "digest": "4BTuRcirHI8zH/wbuPhLERvocqa0+vjKkU4zu4ce4xA="
    }
]
```
